src/module.c

```diff
@@ -3052,11 +3052,10 @@ ValkeyModuleString *moduleAssertUnsharedString(ValkeyModuleString *str) {
     if (str->encoding == OBJ_ENCODING_EMBSTR) {
         /* Note: here we "leak" the additional allocation that was
          * used in order to store the embedded string in the object. */
-        str->ptr = sdsnewlen(str->ptr, sdslen(str->ptr));
-        str->encoding = OBJ_ENCODING_RAW;
+        objectUnembedVal(str, sdsnewlen(objectGetVal(str), sdslen(objectGetVal(str))));
```

Comment:
This is the only place `objectUnembedVal` is called AFAICT.

It'd be cleaner to just pass one argument and let `objectUnembedVal` do the rest.

```suggestion
        objectUnembedVal(str);
```

See also the comment where this function is defined: https://github.com/valkey-io/valkey/pull/2516/files#r2622257872

Reply:
You're right, this is the only place it's called! I was trying to be too type-agnostic in object.c perhaps - I'll implement your suggestion.